presto-hive/src/test/java/io/prestosql/plugin/hive/benchmark/FileFormat.java

```diff
@@ -281,6 +328,85 @@ public abstract FormatWriter createFileFormatWriter(
             HiveCompressionCodec compressionCodec)
             throws IOException;
 
+    public Optional<HivePageSourceFactory> getHivePageSourceFactory(HdfsEnvironment environment)
+    {
+        return Optional.empty();
+    }
+
+    public Optional<HiveRecordCursorProvider> getHiveRecordCursorProvider(HdfsEnvironment environment)
+    {
+        return Optional.empty();
+    }
+
+    public final ConnectorPageSource createFileFormatReader(
```

Comment:
Actually, do we need this method if we have `createGenericReader`? Would `createGenericReader` create page source differently than this method?

> createHiveReader returns a format-specific reader with the hive connector's wrapper page source.

`HivePageSourceProvider` should first try to instantiate native reader

Reply:
> HivePageSourceProvider should first try to instantiate native reader
Currently `HivePageSourceProvider::createPageSource` does that implicitly, when provided with the factories. So simulating that flow requires providing a factory, without instantiating the reader. Not sure if I understand your point correctly.


> Would createGenericReader create page source differently than this method?
The method is the same, the pagesource returned by `createGenericReader` may incur some overhead in addition to the processing done by the delegate readers returned in `createFileFormatReader`. I kept `createFileFormatReader` thinking that the existing benchmarks testing the file-formate specific readers would want numbers without the HivePageSource's overhead.